List.h

```diff
@@ -0,0 +1,255 @@
+#pragma once
+#include <stdlib.h>
+#include <string.h>
+
+void StringListInit(char*** list) {
+	
+	list = nullptr;
+	list = (char***)malloc(2 * sizeof(char**));
+	list[0] = nullptr;
+}
+
+bool IsEmpty(char*** list) {
+	
+	if ((list == nullptr) || (*list == nullptr))
+		return true;
+
+	return false;
+}
+
+size_t StringListSize(char*** list) {
+
+	if (IsEmpty(list))
+		return 0;
+
+	size_t i = 0;
+	char** next = *(list);
+
+	while (next != nullptr) 
+	{
+		++i;
+		next = *(list + i);
+	}
+	return i;
+}
+
+void StringListDestroy(char*** list) {
+	
+	if (IsEmpty(list))
+		return;
+
+	size_t i = 1;
+	char** p = nullptr;
+
+	while (*list != nullptr)
+	{	
+		p = *(list + i);
+		free(*list);
+		*list = p;
+		++i;			
+	}
+	*list = nullptr;
+}
+
+void StringListAdd(char*** list, const char* str) {
+
+	if (list == nullptr)
+		return; 
+
+	if (str == nullptr)
+		return;
+
+	char** elem = nullptr;
+	elem = (char**)malloc(1 * sizeof(char*));
+	elem[0] = (char*)malloc(strlen(str) * sizeof(char));	
+
+	strcpy(elem[0], str);
+
+	if (*list == nullptr) 
+	{
+		list[0] = elem;
+		list[1] = nullptr;
+		return;
+	}
+
+	size_t index = StringListSize(list);
+
+	list[index] = elem;
+	list[index + 1] = nullptr;
+}
+
+size_t StringListIndexOf(char*** list, char* str) {
+	
+	if (IsEmpty(list))
+		return 0;
+
+	if (str == nullptr)
+		return 0;
+
+	const size_t SIZE = StringListSize(list);
+
+	for (size_t i = 0; i != SIZE; ++i)
+	{
+		if (strcmp(*list[i], str) == 0)
+			return i;
+	}
+
+	return SIZE;
```

Comment:
You should check the case when your list haven't such string

Reply:
I check it in main function if return value equals size of my list it means that list hasn"t such string

Reply:
I check it in main function if return value equals size of my list it means that list hasn"t such string